packages/contracts/test/dollar/core/CreditClock.t.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import "../../../src/dollar/core/CreditClock.sol";
+
+import "../../helpers/LocalTestHelper.sol";
+
+contract CreditClockTest is LocalTestHelper {
+    using ABDKMathQuad for uint256;
+    using ABDKMathQuad for bytes16;
+
+    UbiquityDollarManager manager;
+    CreditClock creditClock;
+
+    function setUp() public {
+        manager = new UbiquityDollarManager(address(this));
+        creditClock = new CreditClock(manager, uint256(50).fromUInt(), uint256(3).fromUInt());
+    }
+
+    function testConstructor() public view {
+        require(creditClock.manager() == manager);
+        require(creditClock.rateStartBlock() == block.number);
+        require(creditClock.rateStartValue() == uint256(50).fromUInt());
+        require(creditClock.ratePerBlock() == uint256(3).fromUInt());
+    }
+
+    function testSetRatePerBlock() public {
+        creditClock.setRatePerBlock(uint256(1).fromUInt());
+
+        require(creditClock.rateStartBlock() == block.number);
+        require(creditClock.rateStartValue() == uint256(50).fromUInt());
+        require(creditClock.ratePerBlock() == uint256(1).fromUInt());
```

Comment:
here we are testing against constructor start value and rate per block 
it would be nice to have a test that set rate several time with a few days between each rate change

Reply:
I greatly extended the calculation tests.